sdk/monitor/azure-monitor-query/samples/notebooks/sample_large_query.ipynb

```diff
@@ -0,0 +1,592 @@
+    "    start_time = end_time - timedelta(days=days_back)\n",
+    "    try:\n",
+    "        response = await client.query_workspace(\n",
+    "            workspace_id=LOGS_WORKSPACE_ID,\n",
+    "            query=find_batch_endpoints_query,\n",
+    "            timespan=(start_time, end_time),\n",
+    "        )\n",
+    "    except HttpResponseError as e:\n",
+    "        print(\"Error batching endpoints by row count\")\n",
+    "        raise e\n",
+    "\n",
+    "    batch_endpoints = [end_time]\n",
+    "    batch_endpoints += [row[0] for row in response.tables[0].rows]\n",
+    "    return batch_endpoints\n",
+    "\n",
+    "\n",
+    "async def get_batch_endpoints_by_byte_size(\n",
+    "    query: str,\n",
+    "    end_time: datetime, \n",
+    "    days_back: int,\n",
+    "    max_bytes_per_query: int = 100 * 1024 * 1024, # 100 MiB\n",
+    "    time_col: str = \"TimeGenerated\",\n",
+    "):\n",
+    "    \"\"\"\n",
+    "    Determine the timestamp endpoints for each chunked query such that\n",
+    "    the size of the data returned is less than `max_bytes_per_query`.\n",
+    "    \"\"\"\n",
+    "    \n",
+    "    # This query will assign a batch number to each row depending on the estimated data size.\n",
+    "    # Then the earliest timestamp for each batch number is used for each query endpoint.\n",
+    "    find_batch_endpoints_query = f\"\"\"\n",
+    "        {query}\n",
+    "        | sort by {time_col} desc\n",
+    "        | extend batch_num = row_cumsum(estimate_data_size(*)) / {max_bytes_per_query}\n",
+    "        | summarize endpoint=min({time_col}) by batch_num\n",
+    "        | sort by batch_num asc\n",
+    "        | project endpoint\n",
+    "    \"\"\"\n",
+    "\n",
+    "    start_time = end_time - timedelta(days=days_back)\n",
+    "    try:\n",
+    "        response = await client.query_workspace(\n",
+    "            workspace_id=LOGS_WORKSPACE_ID,\n",
+    "            query=find_batch_endpoints_query,\n",
+    "            timespan=(start_time, end_time)\n",
+    "        )\n",
+    "    except HttpResponseError as e:\n",
+    "        print(\"Error batching endpoints by byte size\")\n",
+    "        raise e\n",
+    "\n",
+    "    batch_endpoints = [end_time]\n",
+    "    batch_endpoints += [row[0] for row in response.tables[0].rows]\n",
+    "    return batch_endpoints"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Next, define a function that will asynchronously execute a given query over a time range specified by a given start time and end time. This function will call the `query_workspace` method of the `LogsQueryClient`. Internally, Azure Core will handle retries in case of connection-related errors or server errors (i.e. 500, 503, and 504 status codes). Check [here](https://github.com/Azure/azure-sdk-for-python/tree/main/sdk/core/azure-core#configurations) for more information on configuring retries."
```

Comment:
We can just say that the monitor query client library itself will handle retries in case of connection-related errors etc. and link out to core as you have :)

Reply:
Good idea, that sounds better.